File: PY/Luciano_Chavarria/Strings/AnagramSearch.py
from sys import stdout,stdin

MAX = 256

def compare(arr1,arr2):
	global MAX
	for i in range(MAX):
		if arr1[i] != arr2[i]:
			return False
	return True

def anagramsSearch(pat,txt):
	global MAX
	M = len(pat)
	N = len(txt)

	countP = [0 for x in range(MAX)]
	countTW = [0 for x in range(MAX)]

	for i in range(M):
		countP[ord(pat[i])]+=1
		countTW[ord(pat[i])]+=1
	for i in range(M,N):
		if compare(countP,countTW):
			stdout.write(f"Encontrado en el indice {i-M} Anagrama {txt[i-M:i]}")
			stdout.write("\n")
		countTW[ord(txt[i])]+=1
		countTW[ord(txt[i-M])]-=1
	if compare(countP,countTW):
		stdout.write(f"Encontrado en el indice {N-M} Anagrama {txt[N-M:N]}")
		stdout.write("\n")



def main():
	txt = "anitalavalatina"
	pat = "anita"
	anagramsSearch(pat,txt)

if __name__ == '__main__':
	main()

File: Codigos estudiantes por lenguaje/Java/Luciano_Chavarria/Strings/AnagramSearch.java
public class Main {

    static final int MAX = 256;

    static boolean compare(char arr1[], char arr2[]) {
        for (int i = 0; i < MAX; i++) {
            if (arr1[i] != arr2[i]) {
                return false;
            }
        }
        return true;
    }

    static void anagramsSearch(String pat, String txt) {
        int M = pat.length();
        int N = txt.length();
        char[] countP = new char[MAX];
        char[] countTW = new char[MAX];
        for (int i = 0; i < M; i++) {
            countP[pat.charAt(i)]++;
            countTW[txt.charAt(i)]++;
        }
        for (int i = M; i <N; i++) {
            if (compare(countP, countTW)) {
                System.out.println("Encontrado en indice " + (i - M)+" Anagrama "+txt.substring(i-M,i));
            }
            countTW[txt.charAt(i)]++;
            countTW[txt.charAt(i - M)]--;
        }
        if (compare(countP, countTW)) {
            System.out.println("Encontrado en indice " + (N - M)+" Anagrama "+txt.substring(N-M,N));
        }
    }

    public static void main(String[] args) {
        String txt= "anitalavalatina";
        String par= "anita";
        anagramsSearch(par, txt);
    }
}
